### PR TITLE
[MM-46678] - Feature Discovery shows Upgrade to Enterprise plan

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
For feature discovery of an enterprise feature, show `Contact sales`

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46678

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
Screenshot 2022-08-31 at 14 38 13

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]
LGTM, thanks @AGMETEOR !

File: components/admin_console/feature_discovery/feature_discovery.tsx [hunk 1]
```diff
@@ -144,6 +149,24 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
...
                 );
+
+                if (minimumSKURequiredForFeature === LicenseSkus.Enterprise) {
+                    ctaPrimaryButton = (
+                        <button
+                            className='btn btn-primary'
+                            data-testid='featureDiscovery_primaryCallToAction'
+                            onClick={() => {
+                                trackEvent(TELEMETRY_CATEGORIES.CLOUD_ADMIN, 'click_enterprise_contact_sales_feature_discovery');
+                                window.open(contactSalesLink, '_blank');
```
Comment (marianunez) [inline components/admin_console/feature_discovery/feature_discovery.tsx:160]: Leaving the comment here as discussed offline, we need to support the contact sales link for the case of self-managed here

Comment (AGMETEOR) [reply components/admin_console/feature_discovery/feature_discovery.tsx]: Just pushed the fix for that

**Review by yasserfaraazkhan — COMMENTED** [review]
Thank you adding tests @AGMETEOR .

I've attached the Zephyr test links to the JIRA MM-46678

File: e2e/cypress/tests/integration/system_console/feature_discovery_cloud_spec.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            is_free_trial: 'false',
+            trial_end_at: 1,
+        };
+        simulateSubscription(subscription);
+        withTrialBefore('false');
+        enterprisePaidFeatures.forEach(({sidebarName, featureDiscoveryTitle}) => {
+            cy.get('li').contains(sidebarName).click();
+            cy.get("div[data-testid='featureDiscovery_title']").should('contain', featureDiscoveryTitle);
+            cy.get("button[data-testid='featureDiscovery_primaryCallToAction']").should('contain', 'Contact sales');
+        });
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply e2e/cypress/tests/integration/system_console/feature_discovery_cloud_spec.js:113]: I don't think we ever run into that case. When a trial has been done before, the ctas are different

File: e2e/cypress/tests/integration/system_console/feature_discovery_cloud_spec.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    it('Ensure feature discovery for enterprise features shows option to contact sales', () => {
+        const subscription = {
+            id: 'sub_test1',
+            product_id: 'prod_1',
+            is_free_trial: 'false',
+            trial_end_at: 1,
+        };
+        simulateSubscription(subscription);
+        withTrialBefore('false');
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply e2e/cypress/tests/integration/system_console/feature_discovery_cloud_spec.js:108]: For cloud, we never ever run into that scenario before of there being a trial license before. For cloud, the scenario you mean is there being a trial subscription before.

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**